I ran this PR's head on GPU (RTX PRO 6000, CUDA 12.9) since CI only exercises the CPU backend — the two CUDA test failures from #1503 are still present, though the margin one gets much closer:

```
FAILED GJKTest::test_sphere_mesh_margin   -0.0009959713 != -0.001 within 7 places  (was 7.1mm off on main)
FAILED GJKTest::test_box_box_ccd          1 contact instead of 4
```

Recomputing `xnorm` together with `x_k` fixes the big error (on main, `result.dist` was one iteration staler than the witness points whenever the loop exhausted its iterations). What remains is the renormalization block: the barycentric coordinates already sum to one by construction (`lambda_i = C4_i / m_det` with `m_det` the cofactor sum), so the block only redistributes rounding residuals into the largest coordinate — but doing that every iteration perturbs the iterate enough on CUDA that the Frank-Wolfe gap never meets its tolerance. Instrumenting the exits on the sphere-mesh case: without the renormalization GJK terminates via the FW gap at iteration 4 with the exact answer; with it, the loop runs all 35 iterations and lands 4e-6 off, and the perturbed witness points also degrade the box-box multiccd manifold (the 1-vs-4 contact failure). I also checked the coupling directly: with fresh distance reporting and working exit criteria in place, restoring just the renormalization still fails these two tests on CUDA.

#1511 fixes the same failures from the other end (removes the renormalization, scales the no-progress check for float32 — with mjMINVAL it can only fire on bitwise equality in float32, which is also why these failures are CUDA-only) but doesn't have the norm-vs-squared-tolerance fix from this PR. Both fixes look relevant: the criterion scaling here, and the renormalization/no-progress changes in #1511 to make the CUDA suite pass.

--- a/mujoco_warp/_src/collision_gjk.py
+++ b/mujoco_warp/_src/collision_gjk.py
@@ -701,8 +701,6 @@ def gjk(
 
     # remove vertices from the simplex no longer needed
     n = int(0)
-    lmbdamax = float(-1.0)
-    imax = int(-1)
     for i in range(4):
       if lmbda[i] == 0.0:
         continue
@@ -713,8 +711,6 @@ def gjk(
       simplex_index1[n] = simplex_index1[i]
       simplex_index2[n] = simplex_index2[i]
       lmbda[n] = lmbda[i]
-      imax = wp.where(lmbda[n] > lmbdamax, n, imax)
-      lmbdamax = wp.where(lmbda[n] > lmbdamax, lmbda[n], lmbdamax)
       n += int(1)
 
     # SHOULD NOT OCCUR
@@ -725,14 +721,6 @@ def gjk(
     if n == 4:
       xnorm = 0.0
       break
-
-    # clamp barycentric coordinates to an exact partition of unity
-    lmbda[imax] = 1.0
-    acc = float(0.0)
-    for i in range(n):
-      if i != imax:
-        acc += lmbda[i]
-    lmbda[imax] -= acc
 
     # get the next iteration of x_k
     x_k = _linear_combine(n, lmbda, simplex)

--- a/mujoco_warp/_src/collision_gjk.py
+++ b/mujoco_warp/_src/collision_gjk.py
@@ -649,9 +649,8 @@ def gjk(
   simplex_index2 = wp.vec4i()
   n = int(0)
   lmbda = wp.vec4(1.0, 0.0, 0.0, 0.0)  # barycentric coordinates
-  tol2 = tolerance * tolerance
-  epsilon = wp.where(is_discrete, 0.0, 0.5 * tol2)
-  min_norm2 = wp.where(is_discrete, MINVAL2, tol2)
+  epsilon = wp.where(is_discrete, 0.0, 0.5 * tolerance * tolerance)
+  min_norm = wp.where(is_discrete, MINVAL, tolerance)
 
   # set initial guess
   x_k = x1_0 - x2_0
@@ -660,7 +659,7 @@ def gjk(
   xnorm_prev = float(0.0)
 
   for _ in range(gjk_iterations):
-    if xnorm2 < min_norm2 and wp.abs(xnorm_prev - xnorm) < tolerance:
+    if xnorm < min_norm or wp.abs(xnorm_prev - xnorm) < tolerance:
       break
 
     # compute the support point with direction tuning

